crates/rt/src/thread.rs

```diff
@@ -16,25 +15,30 @@ pub fn wait_for_completion() {
         );
     }
 
-    const MAX_THREADS: usize = 32;
     let cur_thread_id = unsafe { sys::furi_thread_get_current_id() };
     let app_id = unsafe { CStr::from_ptr(sys::furi_thread_get_appid(cur_thread_id)) };
-    let mut thread_ids: [sys::FuriThreadId; MAX_THREADS] = [ptr::null_mut(); MAX_THREADS];
+    let furi_thread_list = unsafe { sys::furi_thread_list_alloc() };
 
     'outer: loop {
-        let thread_count =
-            unsafe { sys::furi_thread_enumerate(thread_ids.as_mut_ptr(), MAX_THREADS as u32) }
-                as usize;
+        let thread_count = unsafe { sys::furi_thread_list_size(furi_thread_list) };
 
-        for &thread_id in thread_ids[..thread_count].iter() {
-            let thread_app_id = unsafe { CStr::from_ptr(sys::furi_thread_get_appid(thread_id)) };
+        for thread_index in 0..thread_count {
+            let thread = unsafe { sys::furi_thread_list_get_at(furi_thread_list, thread_index) };
+
+            if thread.is_null() {
+                break;
+            }
+
+            let thread_item = unsafe { *(thread) };
+            let thread_app_id = unsafe { CStr::from_ptr(thread_item.app_id) };
```

Comment:
Given that this conversion is not no-op (it is $O(n)$ with $n$ being the length of the string) I would suggest moving this evaluation after the evaluation and check of `thread_id` so that you can early `continue` by the latter without computing the value of `thread_app_id`.

Reply:
Same, with the latest wrappers they are built inline so hope that's a better solution.